engine/classes/ElggObject.php

```diff
@@ -150,13 +150,19 @@ protected function update() {
 		if (!parent::update()) {
 			return false;
 		}
+
+		update_river_access_by_object($this->guid, $this->access_id);
```

Comment:
Yep: https://github.com/Elgg/Elgg/pull/6203

So perhaps changes related to river should be reverted? And then we should officially deprecate `access_id` column in another PR.